### DISCUSSION TITLE
Create a `sphinx-socialposts` extension?

### DISCUSSION THREAD

**Discussion by choldgraf** [executablebooks/meta#1001] [Ideas]
I recently [wrote a blog post about some functionality in my blog](https://chrisholdgraf.com/blog/2023/social-directive/) that lets you embed social media posts into Sphinx sites. A few folks reached out and asked if this could be made into a pip-installable extension. In particular, @12rambau and @kolibril13.

So I'm curious if others would find it useful to have an extension like this in `executablebooks/`. Would @kolibril13 and/or @12rambau be interested in co-maintaining it? If so I'd be happy to quickly create it and release so that we can iterate. @rowanc1 maybe this is a case where we can experiment with standardizing functionality of an _extension_ across both the python + the javascript parsers?[^1]

[^1]: And a future thought: maybe a directive like ` ```{socialpost} ` could be _replaced_ one day by an embed functionality (e.g. `![](tweet:https://twitter.com/choldgraf/status/1625883290593787904)`.

**Comment by 12rambau** [comment]
I have a cookiecutter for sphinx extention, if people agree I can make set it in no time and hand-over the repository either here or in sphinx-contrib

**Reply by choldgraf** [reply]
@12rambau +1 from me, that would be great

**Reply by 12rambau** [reply]
I'll do that tonight after work and add you to the maintainers. "sphinx-social" is available on pypi are you ok with this name ?

**Reply by choldgraf** [reply]
+1 and thanks 🙂

Open to suggestions on improvements to either API (eg directive name) or implementation / social networks supported

**Reply by kolibril13** [reply]
Thanks for the ping, I'm interested in co-maintaining this package. 
sphinx-social sounds good!

**Reply by 12rambau** [reply]
the repository is up and running, I'll do a first round of dev and add you there afterward: https://github.com/12rambau/sphinx-social

**Reply by 12rambau** [reply]
I just invited you to the repository. it's very clumsy at this stage but we can discuss there some design descisions.

**Reply by choldgraf** [reply]
woot! if we want to add this to executablebooks, you might have to be the one to do it. In that case, I might have to give you temporary `admin` status so that you can do it. I also saw you noted `sphinx-contrib` as well. As long as they are willing to give out admin rights on the repository that we'd add there, I'm happy to do that too, either works for me 👍

**Reply by 12rambau** [reply]
I'm already part of sphinx-contrib so I can add anything already. the idea is yours to beggin with so I prefer to let you decide where it should live. anyway, let's wait until there is something working before finding it a home

**Reply by choldgraf** [reply]
I don't have strong opinions - if `sphinx-contrib` is a healthy place and will allow us to have a neutral place to host the code, let's just go with that! @kolibril13 do you have opinions?

**Reply by kolibril13** [reply]
`sphinx-contrib` sounds great!
